external/model-preparation-algorithm/mpa_tasks/apis/detection/task.py

```diff
@@ -202,36 +208,69 @@ def _init_test_data_cfg(self, dataset: DatasetEntity):
+                        coords /= np.array([width, height, width, height], dtype=float)
+                        coords = np.clip(coords, 0, 1)
+
+                        if probability < confidence_threshold:
+                            continue
+
+                        assigned_label = [ScoredLabel(self._labels[label_idx],
+                                                      probability=probability)]
+                        if coords[3] - coords[1] <= 0 or coords[2] - coords[0] <= 0:
+                            continue
+
+                        shapes.append(Annotation(
+                            Rectangle(x1=coords[0], y1=coords[1], x2=coords[2], y2=coords[3]),
+                            labels=assigned_label))
+            elif self._task_type in {TaskType.INSTANCE_SEGMENTATION, TaskType.ROTATED_DETECTION}:
+                for label_idx, (boxes, masks) in enumerate(zip(*all_results)):
+                    for mask, probability in zip(masks, boxes[:, 4]):
+                        mask = mask.astype(np.uint8)
+                        probability = float(probability)
+                        contours, hierarchies = cv2.findContours(mask, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE)
+                        if hierarchies is None:
+                            continue
+                        for contour, hierarchy in zip(contours, hierarchies[0]):
+                            if hierarchy[3] != -1:
+                                continue
+                            if len(contour) <= 2 or probability < confidence_threshold:
+                                continue
+                            if self._task_type == TaskType.INSTANCE_SEGMENTATION:
+                                points = [Point(x=point[0][0] / width, y=point[0][1] / height) for point in contour]
+                            else:
+                                box_points = cv2.boxPoints(cv2.minAreaRect(contour))
+                                points = [Point(x=point[0] / width, y=point[1] / height) for point in box_points]
+                            labels = [ScoredLabel(self._labels[label_idx], probability=probability)]
+                            polygon = Polygon(points=points)
+                            if cv2.contourArea(contour) > 0 and polygon.get_area() > 1e-12:
+                                shapes.append(Annotation(polygon, labels=labels, id=ID(f"{label_idx:08}")))
+            else:
+                raise RuntimeError(
+                    f"MPA results assignment not implemented for task: {self._task_type}")
 
             dataset_item.append_annotations(shapes)
 
-            if fmap is not None:
-                active_score = TensorEntity(name="representation_vector", numpy=fmap)
+            if feature_vector is not None:
+                active_score = TensorEntity(name="representation_vector", numpy=feature_vector)
                 dataset_item.append_metadata_item(active_score)
 
+    def _patch_data_pipeline(self):
```

Comment:
I added a function to update the data part of the recipe if 'coco_ote_pipeline.py' exists in the model config folder. This is the method used because there is a difference in performance between resnet50 and efficientb2b data pipeline. 

If you have a better idea, please leave a comment.

Reply:
I don't think this might be not a problem because previous detection data pipeline also manually tuned. 

But I have a question about naming convention, segmentation templates have 'data_pipeline.py' instead of specific the name of dataset. Even though detection task is using coco dataset class, but if this can be generalized for wide field, I think 'data_pipeline.py' naming can be better than 'coco_data_pipeline.py'.

How do you think about this?

Reply:
In the case of naming, it is the same as the name of the file used by the ote. The names of the data configurations in the model configurations are all the same.